r/R/filesystem.R

```diff
@@ -418,10 +418,12 @@ SubTreeFileSystem$create <- function(base_path, base_fs = NULL) {
 `$.SubTreeFileSystem` <- function(x, name, ...) {
   # This is to allow delegating methods/properties to the base_fs
   assert_that(is.string(name))
-  if (name %in% ls(x)) {
+  if (name %in% ls(envir = x)) {
     get(name, x)
-  } else {
+  } else if (name %in% ls(envir = x$base_fs)) {
     get(name, x$base_fs)
+  } else {
+    NULL
```

Comment:
Using `envir` in these `ls()` calls stops this cryptic warning from being issued when the external pointer breaks:
```
In addition: Warning message:
In ls(x$base_fs) : ‘x$base_fs’ converted to character string
```
That warning comes from a part of the code in `ls()` that doesn't run if you pass the object as `envir`.